sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -169,29 +174,38 @@ public void upload(InputStream data, long length, boolean overwrite) {
     public void uploadWithResponse(InputStream data, long length, ParallelTransferOptions parallelTransferOptions,
         BlobHttpHeaders headers, Map<String, String> metadata, AccessTier tier, BlobRequestConditions requestConditions,
         Duration timeout, Context context) {
+        uploadWithResponse(data, length, new BlobParallelUploadOptions()
+            .setParallelTransferOptions(parallelTransferOptions).setHeaders(headers).setMetadata(metadata).setTier(tier)
+        .setRequestConditions(requestConditions), timeout, context);
+    }
+
+    /**
+     * Creates a new blob, or updates the content of an existing blob.
+     * <p>
+     * To avoid overwriting, pass "*" to {@link BlobRequestConditions#setIfNoneMatch(String)}.
+     *
+     * @param data The data to write to the blob.
+     * @param length The exact length of the data. It is important that this value match precisely the length of the
+     * data provided in the {@link InputStream}.
+     * @param options {@link BlobParallelUploadOptions}
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     */
+    public Response<BlockBlobItem> uploadWithResponse(InputStream data, long length, BlobParallelUploadOptions options,
```

Comment:
Should a non-`Response` overload be added to match the async client more closely.

Reply:
So an overload that takes an options bag and returns a BlockBlobItem? I can see the value of that. I think if we go that route maybe we should deprecate the overloads that don't take an options bag.